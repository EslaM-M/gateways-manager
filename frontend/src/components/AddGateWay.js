import {
  Box,
  Button,
  FormControl,
  Input,
  InputLabel,
  Modal,
  Typography,
} from "@mui/material";
import Backdrop from "@mui/material/Backdrop";
import React, { useState } from "react";
import IpAddressComponent from "./IpAddressComponent";
const AddGateWay = ({ open, handleClose, handleSubmit }) => {
  const [name, setName] = useState("");
  const [ipAddress, setIpAddress] = useState("");
  const submitForm = (event) => {
    event.preventDefault();
    const data = { name, ipAddress };
    setName("");
    setIpAddress("");
    handleSubmit(data);
  };
  return (
    <Modal
      open={open}
      onClose={handleClose}
      aria-labelledby="modal-add-gateway"
      aria-describedby="modal-add-gateway"
      BackdropComponent={Backdrop}
      BackdropProps={{
        timeout: 500,
      }}
    >
      <Box
        component="form"
        onSubmit={submitForm}
        sx={{
          position: "absolute",
          top: "50%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          width: 400,
          bgcolor: "background.paper",
          border: "2px solid #000",
          boxShadow: 24,
          p: 4,
        }}
      >
        <Typography variant="h5" sx={{ paddingBottom: "40px" }}>
          Add New Gateway
        </Typography>
        <Box
          sx={{
            display: "grid",
            gridTemplateColumns: { sm: "1fr 1fr" },
            gap: 4,
          }}
        >
          <FormControl variant="standard">
            <InputLabel htmlFor="gateway-name">Name</InputLabel>
            <Input
              value={name}
              placeholder="Name"
              onChange={(event) => {
                setName(event.target.value);
              }}
              name="gateway-name"
              id="gateway-name"
            />
          </FormControl>
          <FormControl variant="standard">
            <InputLabel htmlFor="ip-address">Ip Address</InputLabel>
            <Input
              value={ipAddress}
              name="ip-address"
              id="ip-address"
              onChange={(event) => {
                setIpAddress(event.target.value);
              }}
              inputComponent={IpAddressComponent}
            />
          </FormControl>
        </Box>

        <Button
          disabled={!name || !ipAddress}
          type="submit"
          sx={{ marginTop: "30px" }}
          variant="contained"
        >
          Save
        </Button>
      </Box>
    </Modal>
  );
};

export default AddGateWay;
