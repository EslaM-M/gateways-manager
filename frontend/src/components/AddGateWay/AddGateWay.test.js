import { render } from "@testing-library/react";
import AddGateWay from "./index";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
test("Component snapshot render correctly", () => {
  const component = render(
    <AddGateWay open={true} handleClose={() => {}} handleSubmit={() => {}} />
  );
  expect(component).toMatchSnapshot();
});

test("Component Save Button Is Disabled", () => {
  const { getByText } = render(
    <AddGateWay open={true} handleClose={() => {}} handleSubmit={() => {}} />
  );
  expect(getByText("Save")).toBeDisabled();
});

test("Component Save Button Is Enabled if the user enter the required data", async () => {
  const { queryByText, getByTestId } = render(
    <AddGateWay open={true} handleClose={() => {}} handleSubmit={() => {}} />
  );
  const ipAddress = getByTestId("ip-address");
  const Name = getByTestId("gateway-name");
  userEvent.type(ipAddress, "192.168.1.1");
  userEvent.type(Name, "Vendor Name");
  expect(queryByText("Save")).toBeEnabled();
});
