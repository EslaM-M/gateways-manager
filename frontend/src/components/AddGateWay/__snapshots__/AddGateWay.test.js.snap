// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-describedby="modal-add-gateway"
      aria-labelledby="modal-add-gateway"
      class="MuiModal-root css-79ws1d-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <form
        class="MuiBox-root css-1wnsr1i"
        tabindex="-1"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-lnxenn-MuiTypography-root"
        >
          Add New Gateway
        </h5>
        <div
          class="MuiBox-root css-nordtu"
        >
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="gateway-name"
            >
              Name
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                data-testid="gateway-name"
                id="gateway-name"
                name="gateway-name"
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="ip-address"
            >
              Ip Address
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                data-testid="ip-address"
                id="ip-address"
                name="ip-address"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-mlws19-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Save
        </button>
      </form>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
